lib/discordrb/webhooks/modal.rb

```diff
@@ -5,76 +5,197 @@ class Discordrb::Webhooks::Modal
   # A mapping of names to types of components usable in a modal.
   COMPONENT_TYPES = {
     action_row: 1,
-    text_input: 4
+    string_select: 3,
+    text_input: 4,
+    user_select: 5,
+    role_select: 6,
+    mentionable_select: 7,
+    channel_select: 8,
+    text_display: 10,
+    label: 18,
+    file_upload: 19
   }.freeze
 
-  # This builder is used when constructing an ActionRow. All current components must be within an action row, but this can
-  # change in the future. A message can have 5 action rows, each action row can hold a weight of 5. Buttons have a weight of 1,
-  # and dropdowns have a weight of 5.
-  class RowBuilder
-    # A mapping of short names to types of input styles. `short` is a single line where `paragraph` is a block.
+  # This builder is used when adding a label component to a modal.
+  class LabelBuilder
```

Comment:
i wonder if it's worth it for semver purposes so make labelbuilder extend row builder because it would break is_a? checks.. but i also don't know why someone would be doing that necessarily. we should probably strive for keeping our semver contract regardless